pkg/nodejoiner/addnodes.go

```diff
@@ -49,3 +72,72 @@ func saveParams(directory, kubeConfig string) error {
 	}
 	return params.Save(directory)
 }
+
+// InitClients initiates a client.
+func initk8sClient(kubeconfig string) (*kubernetes.Clientset, error) {
```

Comment:
Please reuse the same approach of `ClusterInfo` asset for initializing the client, since it was designed to work both in and outside the cluster (and moreover to be unit-testable)

Reply:
refactored

Reply:
As discussed, ClusterInfo generated 2 clients which is not needed in authConfig asset so we will instead only generate the k8s client as required in authConfig, leaving ClusterInfo initClient unchanged